android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/PackageActivity.java

```diff
@@ -0,0 +1,260 @@
+package com.tavultesoft.kmapro;
+
+import android.annotation.SuppressLint;
+import android.app.ActionBar;
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.view.Gravity;
+import android.view.View;
+import android.view.View.OnClickListener;
+import android.webkit.WebChromeClient;
+import android.webkit.WebSettings;
+import android.webkit.WebView;
+import android.webkit.WebViewClient;
+import android.widget.Button;
+import android.widget.TextView;
+
+import android.util.Log;
+
+import com.tavultesoft.kmea.KMManager;
+import com.tavultesoft.kmea.KeyboardEventHandler;
+import com.tavultesoft.kmea.packages.PackageProcessor;
+import com.tavultesoft.kmea.util.FileUtils;
+
+import java.io.File;
+import java.io.FileFilter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class PackageActivity extends Activity {
+
+  private WebView webView;
+  private AlertDialog alertDialog;
+  private File kmpFile;
+  private File tempPackagePath;
+  private List<Map<String, String>> installedPackageKeyboards;
+  private static ArrayList<KeyboardEventHandler.OnKeyboardDownloadEventListener> kbDownloadEventListeners = null;
+
+  @SuppressLint({"SetJavaScriptEnabled", "InflateParams"})
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    final Context context = this;
+    Bundle bundle = getIntent().getExtras();
+    if (bundle != null) {
+      kmpFile = new File(bundle.getString("kmpFile"));
+    }
+
+    final String pkgId = PackageProcessor.getPackageName(kmpFile);
+    String version = PackageProcessor.getPackageVersion(kmpFile, false);
+    if (version == null || version.isEmpty()) {
+      String message = "Invalid package version in " + kmpFile.getName();
+      showErrorDialog(context, pkgId, message);
```

Comment:
Does the `showErrorDialog` method cancel the rest of the method?  It's not like this method can continue if we have an error...